File: Aula 12 - Atributos e Métodos/amendoim.py
class Insumo:
    def __init__(self, quantidade, mes_validade):
        self.__nome = "Amendoim"
        self.__quantidade = quantidade
        self.__mes_validade = mes_validade

    def esta_valido(self, mes_atual):
        if self.__mes_validade < mes_atual:
            print("ALERTA: Risco de Amendoim Murcho! Produto Vencido")
            return False
        return True

    def exibir_info(self):
        print("\n--- Dados do Insumo ---")
        print("Nome:", self.__nome)
        print("Quantidade:", self.__quantidade)
        print("Mês de validade:", self.__mes_validade)

def ler_mes(mensagem):
    while True:
        try:
            mes = int(input(mensagem))
            if 1 <= mes <= 12:
                return mes
            else:
                print("ERRO: Digite um mês entre 1 e 12.")
        except ValueError:
            print("ERRO: Digite apenas números.")

if __name__ == "__main__":
    print("Cadastro de Amendoim 🥜")

    quantidade = int(input("Digite a quantidade: "))

    mes_validade = ler_mes("Digite o mês de validade (1-12): ")

    amendoim = Insumo(quantidade, mes_validade)

    amendoim.exibir_info()

    mes_atual = ler_mes("\nDigite o mês atual (1-12): ")

    if amendoim.esta_valido(mes_atual):
        print("Amendoim dentro da validade.")
    else:
        print("Amendoim não pode ser utilizado.")

File: Aula 12 - Atributos e Métodos/amendoim.java
import java.util.Scanner;

class Insumo {
    private String nome;
    private int quantidade;
    private int mesValidade;

    public Insumo(int quantidade, int mesValidade) {
        this.nome = "Amendoim";
        this.quantidade = quantidade;
        this.mesValidade = mesValidade;
    }

    public boolean estaValido(int mesAtual) {
        if (this.mesValidade < mesAtual) {
            System.out.println("ALERTA: Risco de Amendoim Murcho! Produto Vencido");
            return false;
        }
        return true;
    }

    public void exibirInfo() {
        System.out.println("\n--- Dados do Insumo ---");
        System.out.println("Nome: " + this.nome);
        System.out.println("Quantidade: " + this.quantidade);
        System.out.println("Mês de validade: " + this.mesValidade);
    }
}

public class Main {

    public static int lerMes(Scanner scanner, String mensagem) {
        while (true) {
            try {
                System.out.print(mensagem);
                int mes = Integer.parseInt(scanner.nextLine());
                if (mes >= 1 && mes <= 12) {
                    return mes;
                } else {
                    System.out.println("ERRO: Digite um mês entre 1 e 12.");
                }
            } catch (NumberFormatException e) {
                System.out.println("ERRO: Digite apenas números.");
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Cadastro de Amendoim 🥜");

        System.out.print("Digite a quantidade: ");
        int quantidade = Integer.parseInt(scanner.nextLine());

        int mesValidade = lerMes(scanner, "Digite o mês de validade (1-12): ");

        Insumo amendoim = new Insumo(quantidade, mesValidade);

        amendoim.exibirInfo();

        int mesAtual = lerMes(scanner, "\nDigite o mês atual (1-12): ");

        if (amendoim.estaValido(mesAtual)) {
            System.out.println("Amendoim dentro da validade.");
        } else {
            System.out.println("Amendoim não pode ser utilizado.");
        }

        scanner.close();
    }
}
